plot-gui-lib/plot/private/gui/snip2d.rkt

```diff
@@ -15,9 +15,18 @@
          "worker-thread.rkt"
          "snip.rkt")
 
-(provide make-2d-plot-snip
-         plot-mouse-event-callback/c
-         2d-plot-snip+c%)
+(define plot-mouse-event-callback/c
+  (-> (is-a?/c snip%) (is-a?/c mouse-event%) (or/c real? #f) (or/c real? #f) any/c))
```

Comment:
Should `(is-a?/c snip%)` be `(is-a?/c 2d-plot-snip+c%)` ?

Reply:
`2d-plot-snip+c%` already depends on `plot-mouse-event-callback/c` and I get a compilation error if I define it like that.  I tried using `letrec`, but that didn't work either:

```racket
(define-values
  (2d-plot-snip%/c plot-mouse-event-callback/c)
  (letrec ([plot-mouse-event-callback/c
            (-> (is-a?/c 2d-plot-snip%/c) (is-a?/c mouse-event%)
                (or/c real? #f) (or/c real? #f)
                any/c)]
           [2d-plot-snip%/c
            (class/c
             (set-mouse-event-callback (->m (or/c plot-mouse-event-callback/c #f) any/c))
             (set-overlay-renderers (->m (or/c (treeof renderer2d?) #f) any/c)))])
    (values 2d-plot-snip%/c plot-mouse-event-callback/c)))
```

If I try to use it I get:

```
2d-plot-snip%/c: undefined;
 cannot use before initialization
```